libraries/abstractions/common_io/test/test_scripts/uart/test_iot_uart_test.py

```diff
@@ -0,0 +1,181 @@
+# Amazon FreeRTOS V1.2.3
+# Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# http://aws.amazon.com/freertos
+# http://www.FreeRTOS.org
+
+
+import serial
+import csv
+import argparse
+import os, sys
+import re
+
+parentdir = os.path.dirname(os.getcwd())
+sys.path.insert(0, parentdir)
+from test_iot_test_template import test_template
+
+
+class uart_test(test_template):
+    """
```

Comment:
Satya suggested that class names usually follow UpperCaseCamelCase convention. Could you update it to TestUartAssisted? The tests I submitted will be updated in the new pr as well.

Reply:
Yes will do this. Saw that a change was made to use Sockets instead of serial, I will need some help understanding / setting up this new flow.

Reply:
Feel free to let me know if you have questions.

Reply:
Great, thank you!